src/payment/instrument/instrument-action-creator.ts

```diff
@@ -131,4 +131,28 @@ export default class InstrumentActionCreator {
             storeId,
         };
     }
+
+    private _getSessionContextWithCurrency(store: ReadableCheckoutStore): SessionContext {
```

Comment:
I think it should be okay to, instead of creating a new method, just modify `_getSessionContext` to include the currency? They look pretty much like the same method to me